frontend/src/components/resourceMap/useQueryParamsState.tsx

```diff
@@ -23,6 +23,9 @@ export function useQueryParamsState<T extends string | undefined>(
 
   // State for managing the value derived from the query parameter
   const [value, setValue] = useState<T | undefined>(() => {
+    if (initialState) {
+      return initialState;
+    }
```

Comment:
AFAIU, if the `initialState` is valid, this logic will never actually get the param from the URL if it has an initialState. Is this intended?
Shouldn't we keep the whole logic but return the `initialState` instead of undefined at the end of this function?

Reply:
rewrote the logic to get rid of useState and it's so much simpler now. also added unit tests